compiler/noirc_evaluator/src/acir/mod.rs

```diff
@@ -2762,6 +2762,13 @@ impl<'a> Context<'a> {
             Intrinsic::FieldLessThan => {
                 unreachable!("FieldLessThan can only be called in unconstrained")
             }
+            Intrinsic::ArrayRefCount | Intrinsic::SliceRefCount => {
+                let zero = self.acir_context.add_constant(FieldElement::zero());
```

Comment:
Why not fail here instead of returning an always-zero value?

Reply:
(not blocking)

Reply:
I'm not sure - after your message I was thinking an error would be a better option so I added one in acir. Then we had:
- acir: error
- brillig: return refcount
- comptime: return 0

which felt odd with the comptime behavior. I thought about erroring in comptime too but I think we should still allow comptime to run code with this function in it. Particularly since `comptime { std::runtime::is_unconstrained()) }` is true. In the end I ended up keeping both acir & comptime at zero but would be open to reintroducing the error.

Reply:
The main reason I thought of for an error would be when testing constrained vs. unconstrained vs. comptime: an error makes it clear that we called one of these intrinsics.